common.go

```diff
@@ -959,6 +967,23 @@ type Certificate struct {
 	Leaf *x509.Certificate
 }
 
+type DelegatedCredential struct {
+	ValidTime int64 // todo - change to Time?
+	PublicKey interface{}
+}
+
+// checks certificate if it contains the DelegationUsage extension
+// required for Delegated Credentials
+func (hs *serverHandshakeState) isCertificateValidForDelegationUsage() bool {
```

Comment:
Got it. The content of the handshake_delegated_cred.go file will be then moved to separate crypto/tls/delcred (or similar name) sub package. This sub package will be responsible for creating, marshalling, signing, validating etc. of the credentials.